src/firefly/java/edu/caltech/ipac/firefly/server/query/lsst/LSSTMultiObjectSearch.java

```diff
@@ -0,0 +1,354 @@
+
+@SearchProcessorImpl(id = "LSSTMultiObjectSearch")
+public class LSSTMultiObjectSearch extends LSSTCataLogSearch{
+    private int timeout  = Integer.parseInt(AppProperties.getProperty("lsst.database.timeoutLimit", "180"));
+
+    @Override
+    protected File loadDataFile(TableServerRequest request) throws IOException, DataAccessException {
+
+
+        try {
+            String catalog = request.getParam("table_name");
+            String filename= request.getParam("filename");
+            DataGroup inDg = getInDataGroup(filename);
+            DataGroup[] dataGroupArray =  getSearchResult(request, inDg);
+            DataGroup dg =  dataGroupArrayToDataGroup(dataGroupArray,inDg, catalog);
+            dg.shrinkToFitData();
+            File outFile = createFile(request, ".tbl");
+            DataGroupWriter.write(outFile, dg);
+            return  outFile;
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new DataAccessException("ERROR:" + e.getMessage(), e);
+        }
+
+    }
+
+    private DataGroup getInDataGroup(String filename) throws ExecutionException, InterruptedException, IOException, EndUserException {
+        boolean badParam = true;
+        if (!StringUtils.isEmpty(filename)) {
+            File uploadFile = ServerContext.convertToFile(filename);
+            if (uploadFile.canRead()) {
+                //can read .tbl or .csv file
+                return  DataGroupReader.readAnyFormat(uploadFile);
+            }
+        }
+        if (badParam) {
+            throw new EndUserException("IRSA search failed, Catalog is unavailable",
+                    "Search Processor did not find the required parameter: " + filename);
+        }
+        return null;
+    }
+
+    private void terminateThreads( ExecutorService executor){
+
+        try {
+            LOGGER.info("attempt to shutdown executor");
+            executor.shutdown();
+            executor.awaitTermination(5, TimeUnit.SECONDS);
+        }
+        catch (InterruptedException e) {
+            LOGGER.info("tasks interrupted");
+        }
+        finally {
+            if (!executor.isTerminated()) {
+                LOGGER.info("cancel non-finished tasks");
+            }
+            executor.shutdownNow();
+            LOGGER.info("shutdown finished");
+        }
+    }
+    /**
+     * This method is calling the concurrent worker to do the search.  it depends on the thread, nThread, to create the
+     * workers.
+     * @param request
+     * @return
+     * @throws ExecutionException
+     * @throws InterruptedException
+     * @throws IOException
+     */
+   private  DataGroup[]  getSearchResult(TableServerRequest request, DataGroup inDg) throws ExecutionException, InterruptedException, IOException, EndUserException, TimeoutException {
+
+
+       DataObject[] rows = inDg.values().toArray(new DataObject[0]);
+
+        int nThread = request.containsParam("nThread")? Integer.parseInt(request.getParam("nThread")) :2;
```

Comment:
i don't think the number of threads should come from the request.  I think is should be property the the search processes keeps